src/main/java/io/mathdojo/useraccountservice/model/requestobjects/AccountRequest.java

```diff
@@ -26,12 +31,27 @@ public AccountRequest() {
      * 
      */
     public AccountRequest(boolean accountVerified, String name, String profileImageLink) {
-        this.accountVerified = accountVerified;
-        this.name = name;
-        this.profileImageLink = profileImageLink;
+        this(accountVerified, name, profileImageLink, UUID.randomUUID().toString());
     }
-
-    public boolean isAccountVerified() {
+    /**
+     * Creates an instance of the AccountRequest class
+     * <p>
+     * 
+     * @param accountVerified  the status of the account being created
+     * @param name             the name of the prospective account holder
+     * @param profileImageLink an image of the profile user to be created
+     * @param id  the id of the account being created
+     * 
+     */
+    public AccountRequest(boolean accountVerified, String name, String profileImageLink, String id) {
+    	
+		this.accountVerified = accountVerified;
+		this.name = name;
+		this.id = id != null ? id : UUID.randomUUID().toString();
```

Comment:
I think the goal should be to remove that logic and enforce every account request must have an id attached to it . I put this in here as a placeholder because I didn't like that null id's were possible. I think it's ok to leave it until we've progressed a bit on the auth0 front and we know the structure of id's of the objects that extend this class look like.

Reply:
I see the concern you're trying to resolve here. Following Domain Driven Development principles though, I still think the more scalable approach is to keep this class as a simple data type and put this logic of safeguarding against persisting User or Organization objects with null Ids in the repository classes or in the Identity Service as those are the major centres of logic with this object. Also, it's worth bearing in mind that when Azure does the initial serialization for us, the id field will often be null because the parameter is actually present in the URL path of the request. We then take this id from the path and create an instance of the necessary subclass using it.

Reply:
now been removed